### ISSUE TITLE
Bug between v0.12.2 to v0.13.0 in git write-back implementation

### ISSUE THREAD

**Issue by abatilo** [argoproj-labs/argocd-image-updater#717]
**Describe the bug**
Image updater fails to update images due to a nil pointer dereference

**To Reproduce**
Upgrade from v0.12.2 to v0.13.0

**Expected behavior**
I'd like my images to continue to get updated.

**Additional context**
My ArgoCD application has the following annotations:
```
  annotations:
    argocd-image-updater.argoproj.io/write-back-method: git
    argocd-image-updater.argoproj.io/write-back-target: kustomization
    argocd-image-updater.argoproj.io/image-list: app=ghcr.io/abatilo/blog
    argocd-image-updater.argoproj.io/app.update-strategy: latest
    argocd-image-updater.argoproj.io/git-branch: main
```
These did NOT change between v0.12.2 and v0.13.0. I did also try switching the update-strategy to `newest-build` but that did not fix anything

**Version**
v0.13.0

**Logs**
```
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:18Z" level=info msg="Committing 1 parameter update(s) for application resumegpt" application=resumegpt
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:18Z" level=info msg="Starting configmap/secret informers"
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg="Configmap/secret informer synced"
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg="secrets informer cancelled"
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg="configmap informer cancelled"
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg="Initializing https://github.com/mentallyanimated/deployments.git to /tmp/git-resumegpt3677827165"
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg="rm -rf /tmp/git-resumegpt3677827165" dir= execID=88171
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater time="2024-05-19T14:18:20Z" level=info msg=Trace args="[rm -rf /tmp/git-resumegpt3677827165]" dir= operation_name="exec rm" time_ms=1.685585
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater panic: runtime error: invalid memory address or nil pointer dereference
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc5c19c]
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater goroutine 771 [running]:
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj/argo-cd/v2/util/git.GitHubAppCreds.Environ({0x39d2e, 0x1bfa89d, {0xc0001c2000, 0x68f}, {0x0, 0x0}, {0xc00086a210, 0x2f}, {0x0, 0x0}, ...})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.8.4/util/git/creds.go:388 +0xa5c
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/ext/git.(*nativeGitClient).runCredentialedCmd(0xc0009442a0, {0x2702576?, 0xc000943260?}, {0xc000b672a8?, 0x6?, 0x2ac15d8?})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/ext/git/client.go:595 +0x8b
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/ext/git.(*nativeGitClient).Fetch(0xc0009442a0, {0x0?, 0x0?})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/ext/git/client.go:332 +0x236
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/pkg/argocd.commitChangesGit(0xc000639800, 0xc0008f1970, {0xc000e98fd8, 0x1, 0x1}, 0x284c910)
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/pkg/argocd/git.go:159 +0x365
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/pkg/argocd.commitChanges(0xc000904ca0?, 0xc000f48e00?, {0xc000e98fd8?, 0xc000961d90?, 0xc00065d668?})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/pkg/argocd/update.go:625 +0x45
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/pkg/argocd.commitChangesLocked(0xc000f99f80?, 0xc0008f1970?, 0x34?, {0xc000e98fd8?, 0x1?, 0x1?})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/pkg/argocd/update.go:607 +0x110
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater github.com/argoproj-labs/argocd-image-updater/pkg/argocd.UpdateApplication(0xc0050c9b60, 0x19?)
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/pkg/argocd/update.go:333 +0x1f2d
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater main.runImageUpdater.func1({_, _}, {{{{0xc0006a55f0, 0xb}, {0xc00095f908, 0x14}}, {{0xc0006a57b6, 0x9}, {0x0, 0x0}, ...}, ...}, ...})
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/cmd/run.go:313 +0x292
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater created by main.runImageUpdater
argocd-image-updater-549d4d97bc-bkxqp argocd-image-updater      /src/argocd-image-updater/cmd/run.go:299 +0xa3d
```

**Comment by ImFlog** [comment]
Same as https://github.com/argoproj-labs/argocd-image-updater/issues/621

**Comment by pasha-codefresh** [comment]
Resolved here: https://github.com/argoproj-labs/argocd-image-updater/pull/723

Fix is available in v0.13.1
